internal/controllers/extension_controller.go

```diff
@@ -0,0 +1,177 @@
+	// Don't do anything if Paused
+	if ext.Spec.Managed == ocv1alpha1.ManagedStatePaused {
+		return ctrl.Result{}, nil
+	}
+
+	// validate spec
+	if err := validators.ValidateExtensionSpec(ext); err != nil {
+		// Set the TypeInstalled condition to Unknown to indicate that the resolution
+		// hasn't been attempted yet, due to the spec being invalid.
+		ext.Status.InstalledBundleResource = ""
+		setInstalledStatusConditionUnknown(&ext.Status.Conditions, "installation has not been attempted as spec is invalid", ext.GetGeneration())
+		// Set the TypeResolved condition to Unknown to indicate that the resolution
+		// hasn't been attempted yet, due to the spec being invalid.
+		ext.Status.ResolvedBundleResource = ""
+		setResolvedStatusConditionUnknown(&ext.Status.Conditions, "validation has not been attempted as spec is invalid", ext.GetGeneration())
+
+		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as spec is invalid", ext.GetGeneration())
+		return ctrl.Result{}, nil
+	}
+
+	// TODO: kapp-controller integration
+	// * gather variables for resolution
+	// * perform resolution
+	// * lookup the bundle in the solution/selection that corresponds to the Extension's desired Source
+	// * set the status of the Extension based on the respective deployed application status conditions.
+
+	// Set the TypeInstalled condition to Unknown to indicate that the resolution
+	// hasn't been attempted yet, due to the spec being invalid.
+	ext.Status.InstalledBundleResource = ""
+	setInstalledStatusConditionUnknown(&ext.Status.Conditions, "the Extension interface is not fully implemented", ext.GetGeneration())
+	// Set the TypeResolved condition to Unknown to indicate that the resolution
+	// hasn't been attempted yet, due to the spec being invalid.
+	ext.Status.ResolvedBundleResource = ""
+	setResolvedStatusConditionUnknown(&ext.Status.Conditions, "the Extension interface is not fully implemented", ext.GetGeneration())
+
+	setDeprecationStatusesUnknown(&ext.Status.Conditions, "the Extension interface is not fully implemented", ext.GetGeneration())
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// TODO: Add watch for kapp-controller resources
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ocv1alpha1.Extension{}).
+		Watches(&catalogd.Catalog{},
+			handler.EnqueueRequestsFromMapFunc(extensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
+		Complete(r)
+}
+
+// Generate reconcile requests for all extensions affected by a catalog change
+func extensionRequestsForCatalog(c client.Reader, logger logr.Logger) handler.MapFunc {
+	return func(ctx context.Context, _ client.Object) []reconcile.Request {
+		// no way of associating an extension to a catalog so create reconcile requests for everything
```

Comment:
Given we are not mapping catalog and extension, and creating reconcile requests for all extensions, I don't think we need this mapping func. Watching all catalogs should be enough (in the current state).

Reply:
I believe this is still necessary as the comment says, there's no way to indicate ownership due to the namespace scope of `Extension`. I can look, but this is how `ClusterExtension` does it.

Reply:
I think I figured out the correct method.